CIP-XXXX/README.md

```diff
@@ -0,0 +1,83 @@
+---
+CIP: TBD
+Title: Implement Ouroboros Leios to increase Cardano throughput
+Authors: Duncan Coutts <duncan.coutts@iohk.io>
+Status: Draft
+Type: Core
+Created: 2022-11-18
+License: CC-BY-4.0
+---
+
+Included documents: [*Ouroboros Leios: design goals and concepts*](leios-design.pdf)
+
+## Abstract
+
+As Cardano evolves, there will be increasing demand for greater network
+capacity to support new and existing users and applications. The long term
+solution is to rebase Cardano on the new Ouroboros Leios protocol.
+Ouroboros Leios is a new member of the Ouroboros family that is designed
+specifically for high throughput, without compromising security.  This will
+meet expected future demands, providing a basis for continuing Cardano growth
+and scalability.
+
+## Motivation
+
+Cardano's current throughput (measured both in data rate and available script
+execution time) is adequate for the current demand. There is also some
+opportunity to increase the block sizes and script execution limits to meet
+emerging future demands for increased network capacity. There are however
+fundamental limits to how far the block size and the script execution budget
+can be pushed, while maintaining system security.
+
+Under Ouroboros Praos, in order to ensure the security of the overall system,
+blocks must be distributed across the network reliably in "$\Delta$" time slots.
+This is set to be 5 seconds on the Cardano mainnet. The block relaying process
+is an essentially serial process: blocks must be relayed between consecutive
+block producer nodes through a series of intermediate relay nodes. The overall
+time that this takes is proportional to the number of network hops between one
+block producer and the next, and the network latency of each of those hops
+(which must in general span the whole globe). Given that this must always
+happen within 5 seconds, this puts a hard upper limit on how large each block
+can be and also on how much time can be spent validating transactions and
+scripts.
+
+In order to substantially scale beyond this requires changes to the underlying
+blockchain algorithm. There are significant opportunities to scale: the
+network and CPU resources on most nodes are almost idle much of the time. With
+a different algorithm, these resources can be used to increase the total chain
+bandwidth.
+
+## Specification
+
+Ouroboros Leios is a substantial new design. To do it justice, we do not
+include it in full in this README. Instead, as part of this CIP we include a
+larger document that describes Ouroboros Leios in much more detail:
+
+[*Ouroboros Leios: design goals and concepts*](leios-design.pdf)
```

Comment:
I like the design of Leios.

Two thoughts:
I'm not sure the colour system is optimal. It seems to me that there might be a way to integrate colours deeper into the ledger to further prevent conflicting transactions in concurrent input blocks. I don't have any concrete suggestions unfortunately.


It seems to me that the concept of endorsement certificates can be entirely replaced by (probablistic) proofs à la ZKPs. This is however more complex because of the still unstable nature of the technology, but it seems useful to note that in a future extension/development, it could be replaced by such a mechanism.

Otherwise LGTM (besides the missing parts of course about e.g. rewards)

Reply:
Thanks for the feedback.

The colour system does still need to be properly prototyped & validated, and if it doesn't work as well as hoped then we need to look for alternatives.

Yes, we've not detailed rewards, but the general idea is that the reward payouts are the same in total, the only difference is for SPOs and is about which blocks we count. Obviously in Praos we simply counted blocks, since there is only one kind of block. For Leios we will need to figure out how to count input blocks, endorsement blocks and ranking blocks, given that they all occur at different frequencies. For example, should the less frequent endorsement and ranking blocks be weighted more than input blocks? This is a detail that needs to be worked out.